test/replication.rest.test.js

```diff
@@ -0,0 +1,484 @@
+var async = require('async');
+var debug = require('debug')('test');
+var extend = require('util')._extend;
+var loopback = require('../');
+var expect = require('chai').expect;
+var supertest = require('supertest');
+
+describe('Replication over REST', function() {
+  var ALICE = { id: 'a', username: 'alice', email: 'a@t.io', password: 'p' };
+  var PETER = { id: 'p', username: 'peter', email: 'p@t.io', password: 'p' };
+  var EMERY = { id: 'e', username: 'emery', email: 'e@t.io', password: 'p' };
+
+  var serverApp, serverUrl, ServerUser, ServerCar, serverCars;
+  var aliceId, peterId, aliceToken, peterToken, emeryToken, request;
+  var clientApp, LocalUser, LocalCar, RemoteUser, RemoteCar, clientCars;
+
+  before(setupServer);
+  before(setupClient);
+  beforeEach(seedServerData);
+  beforeEach(seedClientData);
+
+  describe('scenario under test', function() {
```

Comment:
This is intentional. The test suite is checking that I the before/beforeEach hooks have correctly set up all models and permissions. You would not believe how much are the tests intertwined through the global model registry and how easy it is to mess up the setup.

Can you come up with a better name that will make my intent more clear?